ext/liquid_c/vm.c

```diff
@@ -17,13 +15,16 @@ void liquid_vm_render(block_body_t *body, VALUE context, VALUE output)
     Check_Type(output, T_STRING);
 
     VALUE resource_limits = rb_ivar_get(context, id_ivar_resource_limits);
+    resource_limits_t *resource_limits_struct;
+    ResourceLimits_Get_Struct(resource_limits, resource_limits_struct);
```

Comment:
We shouldn't even need the `resource_limits` object after we get the struct pointer out of it, so we may as well not keep it accessible after this

```suggestion
    resource_limits_t *resource_limits;
    ResourceLimits_Get_Struct(rb_ivar_get(context, id_ivar_resource_limits), resource_limits);
```

Specifically, we want to replace the `rb_funcall(resource_limits, id_increment_write_score, 1, output);` line, since it would be on a hot path within the capture tag.